### ISSUE TITLE
Do not flag for CA1056 on members decorated as `[DataMember]`

### ISSUE THREAD

**Issue by RowlandShaw** [dotnet/roslyn-analyzers#4935]
### Analyzer

**Diagnostic ID**: [CA1056](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1056)

### Describe the improvement

Do not flag for CA1056 on members decorated as `[DataMember]`

### Describe suggestions on how to achieve the rule

Some rules, including [CA2227](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2227)', will not flag on members decorated with the `[DataMember]` attribute. I'd suggest that CA1056 also gain this exclusion, so not to require suppression when building classes that are purely used for data transfer objects.

### Additional context
For example, the [JSON Data Contracts](https://docs.microsoft.com/en-us/bingmaps/rest-services/json-data-contracts) for the Bing Maps API has:

```
    public class ImageryMetadata : Resource  
    {  
        /* omitted for brevity */ 
  
        [DataMember(Name = "imageUrl", EmitDefaultValue = false)]  
        public string ImageUrl { get; set; } // This flags with CA1056  
```

**Comment by mavasani** [comment]
@RowlandShaw Can you please give more details why you feel this exclusion is warranted? One could serialize a URI as a string.

**Comment by sharwell** [comment]
Specifically, even if one serialization framework doesn't understand the serialization of `Uri` objects, another would could be implemented to treat them the same way as a `string`.

**Comment by RowlandShaw** [comment]
I could've sworn that the `DataContractJsonSerializer` didn't know how to handle `Uri` objects, although a few quick checks tells me it can serialise to a string. Maybe I instead need to have a chat with the Bing Maps team, and get their sample updated...

**Comment by Youssef1313** [comment]
@RowlandShaw Are you still having problems with CA1056 for this case?
